### ISSUE TITLE
postuninstall hook

### ISSUE THREAD

**Issue by me-ascii** [bower/bower#1451]
Please
```
"scripts": {
        "postinstall": "grunt",
        "postuninstall": "grunt"
    }
```

**Comment by nickcmaynard** [comment]
Definitely a +1 from me here.  

Wiredep suggests adding a `postinstall` hook to add wiring to HTML files for new packages - it'd be great if there was a corresponding `postuninstall` to enable the removal operation.

**Comment by whyleee** [comment]
:+1:

**Comment by MattDiMu** [comment]
+1
I'm also using the post-installation-hook with bower-requirejs to update the config file. It's not a big issue to have unnecessary dependencies there after an uninstallation, but it's suboptimal...

**Comment by silvenon** [comment]
It would definitely be nice to have this hook. :+1:

**Comment by atropo** [comment]
+1 for me too, it would be great with wiredep

**Comment by sonicparke** [comment]
+1 for me too. Would very nice.

**Comment by PeterKow** [comment]
+1, simple automatisation and keeps things clean :+1:

**Comment by mixmode** [comment]
+1

**Comment by ip2k** [comment]
+1 , esp. for the use-case of wiredep (so it can automatically remove all the script / css tags in your HTML files after Bower uninstalls a thing)

**Comment by ahmadalfy** [comment]
+1, need `postuninstall` hook for `grunt wiredep` too.

**Comment by maccurt** [comment]
Uh duh, yes we need this. If it will inject my css and js when I install something IT SHOULD remove it when I uninstall something. This just seems logical right? Am I missing something? I do appreciate all you do.. Just saying..

**Comment by sheerun** [comment]
It's not as easy as you think to implement. PR with tests is appreciated :)

**Comment by maccurt** [comment]
I understand. There is probably something I don't know about. I totally get that. As a developer I am often asked to do something that is estimated to take only 4 hours, a couple weeks later we found out that really was not as easy as the business owner thought it would be Michael C. McGee (AKA Barnabas)Encouragement  & Strategy DepartmentProfitDreamer.com

**Comment by stylig** [comment]
+1 from me, too
As far as I have seen, there is already a PR https://github.com/bower/bower/pull/1681 maybe that one would need some support to pass the checks?

**Comment by sdvim** [comment]
+1, though I found a temporary workaround using `grunt-contrib-watch`:

```javascript
    watch: {
      bower: {
        files: ['bower.json', '.bowerrc'],
        tasks: ['wiredep']
      }
   }
```

And of course you can add templating tasks after `wiredep`, like `jade` for example, to generate the updated files.

**Comment by thomas-j-moffett** [comment]
+1 for wiredep for me too!

**Comment by mwcz** [comment]
+1 for me, bower-requirejs user.  I have to run bower-requirejs on every grunt build, but I'd much rather run it from a postuninstall hook.

**Comment by anilnatha** [comment]
+1. This hook combined with wiredep would be very helpful when completing the workflow for uninstalling packages via bower.

**Comment by lukelarsen** [comment]
+1 for the postuninstall hook.

**Comment by kringon** [comment]
I'll +1 this aswell -> don't really see how you implement a postinstall hook without a postuninstall :dancer:

**Comment by Jnull** [comment]
I agree, +1 to postuninstall hook, i endorse this.

**Comment by letalumil** [comment]
+1 for the postuninstall hook.

**Comment by Leotomas** [comment]
+1

**Comment by guillermocorrea** [comment]
+1

**Comment by Jimladen** [comment]
+1

**Comment by westonganger** [comment]
+1

**Comment by olso** [comment]
+1

**Comment by SubZero-0** [comment]
+1 :)

**Comment by SubZero-0** [comment]
+1

**Comment by leonardollobato** [comment]
+1

**Comment by sheerun** [comment]
Implemented in #2252 and will be released in bower 1.8
